code/modules/events/dust.dm

```diff
@@ -14,13 +14,17 @@ The "dust" will damage the hull of the station causin minor hull breaches.
 	command_announcement.Announce("The [location_name()] is now passing through a belt of space dust.", "[location_name()] Sensor Array", zlevels = affecting_z)
 
 /datum/event/dust/tick()
-	if(world.time > last_wave + min_delay && prob(10))
+	if(world.time > last_wave + min_delay && prob(10) && length(affecting_z))
 		dust_swarm(severity, affecting_z)
 
 /datum/event/dust/end()
 	command_announcement.Announce("The [location_name()] has now passed through the belt of space dust.", "[location_name()] Sensor Array", zlevels = affecting_z)
 
 /proc/dust_swarm(var/strength = EVENT_LEVEL_MUNDANE, var/list/zlevels)
+
+	if(!length(zlevels))
+		return // Not sure how this happened, but saw it in a runtime on Pyrelight.
```

Comment:
`affecting_z = SSmapping.station_levels`
karzerfeste must not have level data set properly or something, if there were no station levels